packages/frontend/src/app/(sticky-header)/search/page.tsx

```diff
@@ -0,0 +1,64 @@
+import errors from '@twreporter/errors'
+import { LoadMoreResults } from './results'
+import { SearchTitle } from './styled'
+import {
+  getPostOnlySearchResults,
+  transferItemsToPostCards,
+  defaultCount,
+} from '@/app/api/search/route'
+
+const apiKey = process.env.SEARCH_API_KEY || ''
+const cx = process.env.SEARCH_ENGINE_ID || ''
+
+export default async function SearchPage({
+  searchParams,
+}: {
+  searchParams: {
+    q: string
+  }
+}) {
+  if (!searchParams.q) {
+    return <SearchTitle>請輸入要搜尋的字串。</SearchTitle>
+  }
+
+  let data
+  try {
+    data = await getPostOnlySearchResults({
+      q: searchParams.q,
+      apiKey,
+      cx,
+      start: 1,
+      count: defaultCount,
+    })
+  } catch (err) {
+    console.log(
+      JSON.stringify({
+        severity: 'ERROR',
+        message: errors.helpers.printAll(
+          err,
+          { withStack: true, withPayload: true },
+          0,
+          0
+        ),
+      })
+    )
+    return (
+      <SearchTitle>
+        搜尋結果服務異常，請稍候再試。 若持續發生，煩請來信至
+        kidsnews@twreporter.org。
+      </SearchTitle>
+    )
+  }
+
+  const cardItems = transferItemsToPostCards(data?.items)
```

Comment:
could we add error handling for `cardItems`?

Reply:
do you mean to add try catch here?

Reply:
Updated. Please check https://github.com/kids-reporter/kids-reporter-monorepo/pull/150/commits/1cff47315af936c25a24bf630b81a5ad20011d47 commit.

Reply:
oh I mean adding check for returned value `cardItems` in case it was null or undefined

Reply:
Updated. Please see https://github.com/kids-reporter/kids-reporter-monorepo/pull/150/commits/3e2735d099b68aaa58341ae3ee04865a5b4f92fc commit.